src/opamPrinter.mli

```diff
@@ -94,7 +113,53 @@ end
 (** {2 Random utility functions} *)
 
 val value_equals: value -> value -> bool
+[@@ocaml.deprecated "Use OpamPrinter.FullPos.value_equals instead."]
 (** Compares structurally, without considering file positions *)
 
 val opamfile_item_equals: opamfile_item -> opamfile_item -> bool
+[@@ocaml.deprecated "Use OpamPrinter.FullPos.opamfile_item_equals instead."]
 (** Compares structurally, without considering file positions *)
+
+module FullPos : sig
+
+  (** [OpamPrinter] functions, with full position types *)
+
+  open OpamParserTypes.FullPos
+
+  val relop: [< relop_kind ] with_pos -> string
+  val logop: [< logop_kind ] with_pos -> string
+  val pfxop: [< pfxop_kind ] with_pos -> string
+  val env_update_op: env_update_op -> string
+
+  val relop_kind: [< relop_kind ] -> string
+  val logop_kind: [< logop_kind ] -> string
+  val pfxop_kind: [< pfxop_kind ] -> string
+  val env_update_op_kind: env_update_op_kind -> string
+
+  val value : value -> string
+  val value_list: value list with_pos -> string
+  val items: opamfile_item list -> string
+  val opamfile: opamfile -> string
+
+  val format_opamfile: Format.formatter -> opamfile -> unit
+
+  module Normalise : sig
+    val escape_string : string -> string
+    val value : value -> string
+    val item : opamfile_item -> string
+    val item_order : opamfile_item -> opamfile_item -> int
+    val items : opamfile_item list -> string
+    val opamfile : opamfile -> string
+  end
+
+  module Preserved : sig
+    val items: string -> opamfile_item list -> opamfile_item list -> string
+
+    val opamfile: ?format_from:file_name -> opamfile -> string
+  end
+
+  val value_equals: value -> value -> bool
+  val opamfile_item_equals: opamfile_item -> opamfile_item -> bool
```

Comment:
It seems mean (!!) to have the comments on the deprecated functions above - perhaps just duplicate them to here? Equally for documentation, it might be worth putting this module first and then all the deprecated functions below it with a single comment about their oldness!

Reply:
I went into minimal changes, with a full definition for the old definition, and `FullPos` only as a temporary change. But yes, it's better to have the comment in the new one.